lambdas/gro-convert-to-json/src/main/java/uk/gov/di/data/lep/GroConvertToJson.java

```diff
@@ -107,4 +91,28 @@ private String getAuthorisationToken() {
             throw new AuthException("Failed to send authorisation request", e);
         }
     }
+
+    private String convertXmlDataToJson(String xmlData, String authorisationToken) {
+        try {
+            var deathRegistrationGroup = xmlMapper.readValue(xmlData, DeathRegistrationGroup.class);
+            var records = deathRegistrationGroup.deathRegistrations();
+
+            if (records == null) {
+                if(deathRegistrationGroup.recordCount() == 0) {
+                    throw new MappingException("File contains no registration records");
```

Comment:
I'm not entirely convinced this is an exception, thinking about it - it's plausibly just an empty file.

I'm happy enough keeping this though, and if it causes any problems/we get regular empty files then we can always change it later.